cli/pcluster/easyconfig.py

```diff
@@ -63,32 +63,37 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-def prompt(message, default_value=None, hidden=False, options=None, check_validity=False):
-    if hidden and default_value is not None:
-        user_prompt = message + " [*******" + default_value[-4:] + "]: "
-    else:
-        user_prompt = message + " ["
-        if default_value is not None:
-            user_prompt = user_prompt + default_value + "]: "
-        else:
-            user_prompt = user_prompt + "]: "
-
-    if isinstance(options, (list, tuple)):
-        print("Acceptable Values for %s: " % message)
-        for o in options:
-            print("    %s" % o)
-
-    var = input(user_prompt).strip()
-
-    if var == "":
-        return default_value
-    else:
-        if check_validity and options is not None and var not in options:
-            print("ERROR: The value (%s) is not valid " % var)
-            print("Please select one of the Acceptable Values listed above.")
-            sys.exit(1)
+def prompt(message, validator=lambda x: True, input_to_option=lambda x: x, default_value=None, options_to_print=None):
+    """
+    Prompt the user a message with optionally some options.
+
+    :param message: the message to show to the user
+    :param validator: a function that predicates if the input is correct
+    :param input_to_option: a function that given the input transforms it in something else
+    :param default_value: the value to return as the default if the user doesn't insert anything
+    :param options_to_print: the options to print if necessary
+    :return: the value inserted by the user validated
+    """
+    if options_to_print:
+        print("Allowed values for {0}:".format(message))
+        [print(item) for item in options_to_print]
```

Comment:
List comprehensions are used for creating new lists from other iterables. Let's not abuse this construct when not needed